### PR TITLE
Analytics, Optional Features, and Layout updates

### PR DESCRIPTION
**Pull request by JJediny**

### CODE REVIEW

**Comment by waldoj** [comment]
:+1:

File: _config.yml [hunk 1]
```diff
@@ -54,6 +58,30 @@ markdown: kramdown
...
+  subagency: ""
+
+# Hypothesis.io - Annotator
+hypothesis:
+  enabled: true
+
+# Twitter
+twittercard:
+  enabled: false
+  site: ""
```
Comment (timwis) [inline _config.yml]: What does `site` refer to? Doesn't appear to be used by the `addons/twittercard.html` include

**Comment by timwis** [comment]
This is awesome @JJediny! Taking a look now. May ask for one or two things to be broken out into separate PRs if you don't mind

File: _config.yml [hunk 2]
```diff
@@ -1,10 +1,14 @@
 title: JKAN
 description: A lightweight, backend-free open data portal, powered by Jekyll.
 baseurl: /jkan
+logo:
```
Comment (timwis) [inline _config.yml]: Is this intended to be a full `http://`-prefixed URL or a relative path like `/img/logo.png`?

File: _includes/addons/opengraph.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<meta property="og:title" content="{{ site.title }}" />
+<meta property="og:site_name" content="{{ site.title }}" />
+<meta property="og:url" content="{{ site.url }}" />
+<meta property="og:description" content="{{  site.description | truncate: 200 }}" />
+<meta name="og:image" content="{{ site.url }}{{ site.logo }}" />
```
Comment (timwis) [inline _includes/addons/opengraph.html]: Did you mean to use `{{ site.baseurl }}` instead of `url`?

File: _layouts/dataset.html [hunk 4]
```diff
@@ -34,11 +37,11 @@
...
     <ul>
       {% for resource in page.resources %}
-      <li>
-        <a href="{{ resource.url }}">{{ resource.name }}</a>
-        {% if resource.format %}({{ resource.format}}){% endif %}
-        {% if resource.description %}<div class="resource-description">{{ resource.description }}</div>{% endif %}
-      </li>
+        <li>
+          <a href="{{ resource.url }}">{{ resource.name }}</a>
+          {% if resource.format %}<span class="label label-pill label-default">{{ resource.format }}</span>{% endif %}<br>
```
Comment (timwis) [inline _layouts/dataset.html]: Good idea - the labels add a nice touch

File: _layouts/dataset.html [hunk 5]
```diff
@@ -34,11 +37,11 @@
...
+          {% if resource.description %}<div class="resource-description"><code>{{ resource.description }}</code></div>{% endif %}
```
Comment (timwis) [inline _layouts/dataset.html]: May I ask why you went with `<code>` tags for the description rather than adding styles to the current `.resource-description` class in `main.css`?

File: _layouts/dataset.html [hunk 6]
```diff
@@ -15,11 +15,14 @@
...
   {% if organization %}
   <div class="col-sm-3">
+    <div class="panel panel-default">
     {% if organization.logo %}
       <a href="{{ site.baseurl }}/datasets/?organization={{ organization.title | slugify }}" class="thumbnail"><img src="{{ organization.logo }}" alt="{{ organization.title }} logo"></a>
     {% endif %}
-    <h3><a href="{{ site.baseurl }}/datasets/?organization={{ organization.title | slugify }}">{{ organization.title }}</a></h3>
-    <p>{{ organization.description }}</p>
+      <div class="panel-heading"><h3><a href="{{ site.baseurl }}/datasets/?organization={{ organization.title | slugify }}">{{ organization.title }}</a></h3>
+        <div class="panel-body">{{ organization.description }}</p></div>
```
Comment (timwis) [inline _layouts/dataset.html]: Good idea putting this in a panel. But I think you've accidentally put the `panel-body` inside the `panel-heading` rather than as a sibling. Also, I think we should add a conditional to test whether there is a description set, otherwise there's an empty box.

File: _layouts/dataset.html [hunk 7]
```diff
@@ -49,7 +52,7 @@
       <dd>
         {% if page.category.size %}
           {% for category in page.category %}
-            <a href="{{ site.baseurl }}/datasets/?category={{ category | slugify }}">{{ category }}</a><br>
+            <a href="{{ site.baseurl }}/datasets/?category={{ category | slugify }}"><h4><span class="label label-danger">{{ category }}</span></a></h4><br>
```
Comment (timwis) [inline _layouts/dataset.html]: Somehow this seems to be putting large gaps between the categories when there are multiple. Perhaps it's the `<h4>` tag?
screen shot 2016-04-01 at 10 01 05

File: _config.yml (hunk 1 above)
Comment (JJediny) [inline _config.yml]: site here refers to a twitter handle... should change to `handle`?

File: _includes/addons/opengraph.html (hunk 3 above)
Comment (JJediny) [inline _includes/addons/opengraph.html]: yes... will issue a new PR with changes

File: _layouts/dataset.html (hunk 6 above)
Comment (JJediny) [inline _layouts/dataset.html]: will add an if condition...

File: _layouts/dataset.html (hunk 5 above)
Comment (JJediny) [inline _layouts/dataset.html]: try not to add too much in main.css as closer to native bootstrap/jekyll themeing seemed simpler

File: _layouts/dataset.html (hunk 7 above)
Comment (JJediny) [inline _layouts/dataset.html]: Yea i noticed this when testing a dataset with multiple categories... wondering if a button class might be friendlier as that too with render multiple categories inline?

File: _config.yml (hunk 2 above)
Comment (JJediny) [inline _config.yml]: relative path... for the site itself `img/logo.png`

File: _layouts/dataset.html (hunk 4 above)
Comment (JJediny) [inline _layouts/dataset.html]: in CKAN label colors are controlled by the format type, how would you feel about the updated PR adding these to main.css?

**Comment by timwis** [comment]
@JJediny would you mind limiting the scope of this PR to twitter, open graph, google analytics, and the digital analytics program? The style changes should probably be in their own PR (and I think I broke a few of them with #60).

File: _includes/addons/opengraph.html (hunk 3 above)
Comment (timwis) [inline _includes/addons/opengraph.html]: FYI, you can make another commit to your `analytics` branch and github will update this pull request with your latest code

File: _config.yml (hunk 2 above)
Comment (timwis) [inline _config.yml]: Got it -- we should probably add a default logo then, so `_config.yml` can have a default value. Know any logo designers? :P

File: _config.yml (hunk 1 above)
Comment (timwis) [inline _config.yml]: Oh! I think `username` would be the most obvious property name. But it looks like `addons/twittercard.html` doesn't use this property either way.

File: _config.yml (hunk 2 above)
Comment (JJediny) [inline _config.yml]: jkan

?

File: _config.yml (hunk 1 above)
Comment (JJediny) [inline _config.yml]: https://dev.twitter.com/cards/types/summary

`site` in twittercard === twitter handle

**Comment by JJediny** [comment]
removed some of the styling/layout changes from this PR... all others comments should be addressed per the discussion

**Comment by timwis** [comment]
Thanks for all the work on this @JJediny! I made some minor changes (ie. underscores instead of hyphens to match other setting names, fix description in og code, etc.) and merged it.
